File: sdkwork-ai-sdk-typescript/src/types/video-generation-request.ts
/** Video generation request */
export interface VideoGenerationRequest {
  /** Text prompt that describes the video to generate */
  prompt: string;
  /** The video generation model to use */
  model?: 'sora-2' | 'sora-2-pro';
  /** Clip duration in seconds */
  seconds?: '4' | '8' | '12';
  /** Output resolution formatted as width x height */
  size?: '720x1280' | '1280x720' | '1024x1792' | '1792x1024';
  /** Quality of the generated video */
  quality?: string;
  /** Output width in pixels */
  width?: number;
  /** Output height in pixels */
  height?: number;
  /** Seed for reproducible generation */
  seed?: number;
  /** User identifier */
  user?: string;
  /** Negative prompt for video generation */
  negative_prompt?: string;
  /** Image URLs for video generation */
  image_urls?: string[];
  /** Output aspect ratio */
  aspect_ratio?: string;
  /** Output response format */
  response_format?: string;
}

File: sdkwork-ai-sdk-go/types/video_generation_request.go
package types

// Video generation request
type VideoGenerationRequest struct {
	Prompt string `json:"prompt"`
	Model string `json:"model"`
	Seconds string `json:"seconds"`
	Size string `json:"size"`
	Quality string `json:"quality"`
	Width int `json:"width"`
	Height int `json:"height"`
	Seed int `json:"seed"`
	User string `json:"user"`
	NegativePrompt string `json:"negative_prompt"`
	ImageUrls []string `json:"image_urls"`
	AspectRatio string `json:"aspect_ratio"`
	ResponseFormat string `json:"response_format"`
}
